gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java

```diff
@@ -120,40 +112,36 @@ public boolean equals(Object obj) {
         return false;
       }
       Binding other = (Binding) obj;
-      return Objects.equals(name, other.name)
-          && Objects.equals(cursor, other.cursor)
-          && Objects.equals(value, other.value);
+      return Objects.equals(cursor, other.cursor) && Objects.equals(value, other.value);
     }
 
     @Override
-    protected DatastoreV1.GqlQueryArg toPb() {
-      DatastoreV1.GqlQueryArg.Builder argPb = DatastoreV1.GqlQueryArg.newBuilder();
-      if (name != null) {
-        argPb.setName(name);
-      }
+    protected com.google.datastore.v1beta3.GqlQueryParameter toPb() {
+      com.google.datastore.v1beta3.GqlQueryParameter.Builder argPb = 
+          com.google.datastore.v1beta3.GqlQueryParameter.newBuilder();
       if (cursor != null) {
         argPb.setCursor(cursor.byteString());
       }
       if (value != null) {
-        // TODO(ajaykannan): fix me!
-        //argPb.setValue(value.toPb());
+        argPb.setValue(value.toPb());
       }
       return argPb.build();
     }
 
     @Override
     protected Object fromPb(byte[] bytesPb) throws InvalidProtocolBufferException {
-      return fromPb(DatastoreV1.GqlQueryArg.parseFrom(bytesPb));
+      return fromPb(com.google.datastore.v1beta3.GqlQueryParameter.parseFrom(bytesPb));
     }
 
-    static Binding fromPb(DatastoreV1.GqlQueryArg argPb) {
-      String name = argPb.hasName() ? argPb.getName() : null;
-      if (argPb.hasCursor()) {
-        return new Binding(name, new Cursor(argPb.getCursor()));
+    static Binding fromPb(com.google.datastore.v1beta3.GqlQueryParameter argPb) {
```

Comment:
Okay, makes sense, I'll remove the check.  I think the reason I originally asked this was because v1beta3 introduced `TYPE_UNSPECIFIED` for many of it's enums.  @pcostell, we won't get a filter of unspecified type from Datastore either, right?